### DISCUSSION TITLE
Add search and lazy loading of discounts inside CP

### DISCUSSION THREAD

**Discussion by klemenb** [craftcms/commerce#2322] [Ideas] [answered]
### Description

We are planning on generating 10.000 - 100.000 discount codes in the coming months and we are afraid that the current "Discounts" view in the CP will break completely when we do that.

I would suggest that the "Discounts" view gets extended with a search option and lazy loading of elements similar to what is currently available when viewing orders.

Both of these things would come in handy even when there are only ~100 discount codes and I am sure many store owners reach these numbers quite easily.

### Additional info

- Craft version: 3.0.24
- Plugins & versions: Craft Commerce 2.0.0-beta.10

**Comment by lukeholder** [comment]
Guessing you are generating discounts to generate more discount codes?

We are looking to split out codes from discounts. Please keep an eye on this issue.

**Comment by lukeholder** [comment]
- [x] Add pagination
- [x] Add search

**Reply by WHITE-developer** [reply]
This would really help. That page will get really slow with a lot of discounts

**Comment by nfourtythree** [comment, marked as answer]
Commerce version `4.4.0` has now been released which includes pagination and searching of discounts from the discount index.
